### PR TITLE
[DEVX-1553] Add Code field

### PR DESCRIPTION
**Pull request by FriggaHel**
Add code field to be able to capture test code.

### CODE REVIEW

**Review by mhan83 — COMMENTED** [review]
Is the intention here to be able to add the entire test definition to the `code` prop? That seems a bit heavy handed and maybe a little intrusive? If we narrow the scope to failures and errors, I wonder if we want structured data here instead of a single string. At the very least, would we want to associate an error with the code snippet that generated it?

**Comment by FriggaHel** [comment]
@mhan83 Yup, the intention was to be able to attach the code to the test, as Cypress provides it, and then be able to display the actual test-code in the UI (can be compared to selenium commands that are displayed).

While I agree a string is maybe not enough structured for a test code, this is the only way cypress provides that, and currently other frameworks don't look to provide it. It can be enhanced by splittings the lines and/or providing a more complete object that is likely to be more-future proof.

For your last question, I guess that we want. As the test is carrying the information for the test passing/failing, this seems the best location to store that piece of code.

File: src/index.ts [hunk 1]
```diff
@@ -132,16 +134,21 @@ export class Suite {
...
         duration = 0,
         output?: string,
         startTime: Date = new Date(),
         metadata: object = {},
-        attachments: Attachment[] = new Array<Attachment>()
+        attachments: Attachment[] = new Array<Attachment>(),
+        code?: TestCode,
     ): Test {
-        const test = new Test(name, status, duration, output, startTime, attachments, metadata)
+        if (typeof status === "object") {
```
Comment (alexplischke) [inline src/index.ts:145]: Why is that needed? 🤔

**Review by alexplischke — APPROVED** [review]
